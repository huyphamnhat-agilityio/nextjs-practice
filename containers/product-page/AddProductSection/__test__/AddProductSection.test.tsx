import { render, screen } from "@testing-library/react";
import { useDisclosure } from "@nextui-org/react";

// Components
import AddProductionSection from "..";
import userEvent from "@testing-library/user-event";

jest.mock("react-dom", () => ({
  ...jest.requireActual("react-dom"),
  useFormStatus: () => ({ pending: false }),
  useFormState: () => [{}, () => {}],
}));

describe("AddProductionSection test cases", () => {
  // const mockUseDisclosure = useDisclosure as jest.Mock;
  const setup = () => render(<AddProductionSection />);
  it("should render correctly", () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should open the modal when the "Add course" button is clicked', async () => {
    setup();

    const addBtn = screen.getByRole("button", {
      name: /add course/i,
    });

    await userEvent.click(addBtn);

    const modal = screen.getByRole("dialog", {
      name: /add course/i,
    });

    expect(modal).toBeInTheDocument();
  });
});
