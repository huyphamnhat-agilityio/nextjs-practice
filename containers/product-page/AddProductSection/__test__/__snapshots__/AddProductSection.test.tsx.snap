// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProductionSection test cases should render correctly 1`] = `
<DocumentFragment>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-24 text-medium gap-3 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover px-10 py-3.75 w-auto h-auto rounded-1.25 text-white"
    type="button"
  >
    Add course
  </button>
</DocumentFragment>
`;
